lua/null-ls/generators.lua

```diff
@@ -57,6 +57,14 @@ M.run = function(generators, params, opts, callback)
             local ok, results = protected_call(to_run, copied_params)
             a.util.scheduler()
 
+            -- filter diagnostics results with the diagnostic_filter option
+            local filter = opts.filter
```

Comment:
```suggestion
            local filter = generator.opts.filter
```

We can greatly simplify this by simply pulling the filter directly from the generator's options, meaning that we can immediately generalize this beyond just diagnostics.